### PR TITLE
Add EclipsePRTLog backend.

### PR DESCRIPTION
**Pull request by qilicun**
This PR intend to add a prototype for output eclipse PRT file, more usefule functions could be added later.

### CODE REVIEW

File: opm/parser/eclipse/OpmLog/EclipsePRTLog.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    void EclipsePRTLog::clear()
+    {
+        m_count.clear();
+    }
+
+
+    EclipsePRTLog::~EclipsePRTLog()
+    {
+        const auto warning = numMessages(Log::MessageType::Warning);
```
Comment (joakim-hove) [inline opm/parser/eclipse/OpmLog/EclipsePRTLog.cpp]: Not really necessary with these temporary variables?

Comment (qilicun) [inline opm/parser/eclipse/OpmLog/EclipsePRTLog.cpp]: True, will remove them.

File: opm/parser/eclipse/OpmLog/EclipsePRTLog.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            if (iter == m_count.end())
+                return 0;
+            else
+                return (*iter).second;
+        } else
+            throw std::invalid_argument("The messageType ID must be 2^n");
+    }
+
+
+    void EclipsePRTLog::clear()
```
Comment (joakim-hove) [inline opm/parser/eclipse/OpmLog/EclipsePRTLog.cpp]: This method seems like a *might be handy* - method. I have strongly come to the conclusion that such methods should be avoided - *implement it when you need it.*

File: opm/parser/eclipse/OpmLog/EclipsePRTLog.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+*/
+
+#include <opm/parser/eclipse/OpmLog/EclipsePRTLog.hpp>
+#include <opm/parser/eclipse/OpmLog/LogUtil.hpp>
+
+namespace Opm {
+
+
+    EclipsePRTLog::EclipsePRTLog(const std::string& logFile, int64_t messageMask) : StreamLog(logFile, messageMask)
+    {}
```
Comment (joakim-hove) [inline opm/parser/eclipse/OpmLog/EclipsePRTLog.cpp]: As long as these constructors only call the base constructor you do not need them?

Comment (qilicun) [inline opm/parser/eclipse/OpmLog/EclipsePRTLog.cpp]: Does C++ will call the base constructor when you want to create an EclipsePRTLog object?

File: opm/parser/eclipse/OpmLog/EclipsePRTLog.cpp (hunk 2 above)
Comment (qilicun) [inline opm/parser/eclipse/OpmLog/EclipsePRTLog.cpp]: Ok.

File: opm/parser/eclipse/OpmLog/EclipsePRTLog.cpp (hunk 3 above)
Comment (atgeirr) [inline opm/parser/eclipse/OpmLog/EclipsePRTLog.cpp]: Inheriting constructors is a fairly new feature in C++ though, and you must explicitly declare them I think.

Comment (jokva) [inline opm/parser/eclipse/OpmLog/EclipsePRTLog.cpp]: You can delegate to the parent constructor with

`class Foo : public Bar { public: using Bar::Bar; };`

Comment (qilicun) [inline opm/parser/eclipse/OpmLog/EclipsePRTLog.cpp]: Good solution.

**Comment by joakim-hove** [comment]
Thank you.

**Pull request merged**